content/blog/2019/multicluster-version-routing/index.md

```diff
@@ -0,0 +1,455 @@
+    spec:
+      containers:
+      - name: reviews
+        image: istio/examples-bookinfo-reviews-v2:1.10.0
+        imagePullPolicy: IfNotPresent
+        ports:
+        - containerPort: 9080
+---
+apiVersion: extensions/v1beta1
+kind: Deployment
+metadata:
+  name: reviews-v3
+spec:
+  replicas: 1
+  template:
+    metadata:
+      labels:
+        app: reviews
+        version: v3
+    spec:
+      containers:
+      - name: reviews
+        image: istio/examples-bookinfo-reviews-v3:1.10.0
+        imagePullPolicy: IfNotPresent
+        ports:
+        - containerPort: 9080
+EOF
+{{< /text >}}
+
+## Access the `bookinfo` application
+
+Just like any application, we'll use an Istio gateway to access the `bookinfo` application.
+
+* Create the `bookinfo` gateway in `cluster1`:
+
+    {{< text bash >}}
+    $ kubectl apply --context=$CTX_CLUSTER1 -f samples/bookinfo/networking/bookinfo-gateway.yaml
+    {{< /text >}}
+
+* Follow the [Bookinfo sample instructions](/docs/examples/bookinfo/#determining-the-ingress-ip-and-port)
+    to determine the ingress IP and port and then point your browser to `http://$GATEWAY_URL/productpage`.
+
+You should see the `productpage` with reviews, but without ratings, because only `v1` of the `reviews` service
+is running on `cluster1` and we have not yet configured access to `cluster2`.
+
+## Create a service entry and destination rule on `cluster1` for the remote reviews service
+
+As described in the [setup instructions](/docs/setup/kubernetes/multicluster-install/gateways/#setup-dns),
+remote services are accessed with a `.global` DNS name. In our case, it's `reviews.default.global`,
+so we need to create a service entry and destination rule for that host.
+The service entry will use the `cluster2` gateway as the endpoint address to access the service.
+You can use the gateway's DNS name, if it has one, or its public IP, like this:
+
+{{< text bash >}}
+$ export CLUSTER2_GW_ADDR=$(kubectl get --context=$CTX_CLUSTER2 svc --selector=app=istio-ingressgateway \
+    -n istio-system -o jsonpath="{.items[0].status.loadBalancer.ingress[0].ip}")
+{{< /text >}}
+
+Now create the service entry and destination rule using the following command:
+
+{{< text bash >}}
+$ kubectl apply --context=$CTX_CLUSTER1 -f - <<EOF
+apiVersion: networking.istio.io/v1alpha3
+kind: ServiceEntry
+metadata:
+  name: reviews-default
+spec:
+  hosts:
+  - reviews.default.global
+  location: MESH_INTERNAL
+  ports:
+  - name: http1
+    number: 9080
+    protocol: http
+  resolution: DNS
+  addresses:
+  - 127.255.0.3
+  endpoints:
+  - address: ${CLUSTER2_GW_ADDR}
+    labels:
+      version: v2
+      version: v3
+    ports:
+      http1: 15443 # Do not change this port value
+---
+apiVersion: networking.istio.io/v1alpha3
+kind: DestinationRule
+metadata:
+  name: reviews-global
+spec:
+  host: reviews.default.global
+  trafficPolicy:
+    tls:
+      mode: ISTIO_MUTUAL
+  subsets:
+  - name: v1
+    labels:
+      version: v1
```

Comment:
you dont need v1 here right?

Reply:
True, but I figured it was harmless.
I'll remove it.